src/core/Solana/sendAndConfirmTransaction.ts

```diff
@@ -41,71 +42,86 @@ export async function sendAndConfirmTransaction(
     preflightCommitment: 'confirmed',
   }
 
-  for (const connection of connections) {
-    connection
-      .sendRawTransaction(signedTxSerialized, rawTransactionOptions)
-      .catch()
-  }
-
-  const abortControllers: AbortController[] = []
-
-  const confirmPromises = connections.map(async (connection) => {
-    const abortController = new AbortController()
-    abortControllers.push(abortController)
-    try {
-      const blockhashResult = await connection.getLatestBlockhash('confirmed')
-
-      const confirmTransactionPromise = connection
-        .confirmTransaction(
-          {
-            signature: txSignature,
-            blockhash: blockhashResult.blockhash,
-            lastValidBlockHeight: blockhashResult.lastValidBlockHeight,
-            abortSignal: abortController.signal,
-          },
-          'confirmed'
-        )
-        .then((result) => result.value)
-
-      let signatureResult: SignatureResult | null = null
-      let blockHeight = await connection.getBlockHeight('confirmed')
+  const pollPromises = connections.map(async (connection) => {
+    const timeout = 60000
+    const startTime = Date.now()
+    let sentSignature: string | null = null
 
-      while (
-        !signatureResult &&
-        blockHeight < blockhashResult.lastValidBlockHeight
-      ) {
-        await connection.sendRawTransaction(
+    while (Date.now() - startTime < timeout) {
+      try {
+        sentSignature = await connection.sendRawTransaction(
           signedTxSerialized,
           rawTransactionOptions
         )
-        signatureResult = await Promise.race([
-          confirmTransactionPromise,
-          sleep(1000),
-        ])
 
-        if (signatureResult || abortController.signal.aborted) {
-          break
+        const confirmedSignature = await pollTransactionConfirmation(
+          sentSignature,
+          connection
+        )
+        return {
+          signatureResult: { err: null },
+          txSignature: confirmedSignature,
         }
-
-        blockHeight = await connection.getBlockHeight('confirmed')
+      } catch (error) {
+        console.warn('Failed to send transaction to connection:', error)
       }
+    }
 
-      abortController.abort()
+    throw new Error('Failed to send transaction after timeout')
+  })
+
+  try {
+    const result = await Promise.any(pollPromises)
+    return result
+  } catch (error) {
+    throw new Error(
+      `Failed to send and confirm transaction on any connection: ${error}`
+    )
+  }
+}
 
-      return signatureResult
+async function pollTransactionConfirmation(
+  txtSig: TransactionSignature,
+  connection: Connection
+): Promise<TransactionSignature> {
+  // 15 second timeout
+  const timeout = 15000
+  // 5 second retry interval
+  const interval = 5000
+  const startTime = Date.now()
```

Comment:
Same here. It would be nice to keep 1-second timeout for this to speed up transaction landing.